pkg/provider/kubernetes/ingress/client_test.go

```diff
@@ -190,6 +191,102 @@ func TestClientIgnoresHelmOwnedSecrets(t *testing.T) {
 	assert.False(t, found)
 }
 
+func TestClientIgnoresEmptyEndpointUpdates(t *testing.T) {
+	emptyEndpoint := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "empty-endpoint",
+			Namespace:       "test",
+			ResourceVersion: "1244",
+			Annotations: map[string]string{
+				"test-annotation": "_",
+			},
+		},
+	}
+
+	filledEndpoint := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "filled-endpoint",
+			Namespace:       "test",
+			ResourceVersion: "1234",
+		},
+		Subsets: []corev1.EndpointSubset{{
+			Addresses: []corev1.EndpointAddress{{
+				IP: "10.13.37.1",
+			}},
+			Ports: []corev1.EndpointPort{{
+				Name:     "testing",
+				Port:     1337,
+				Protocol: "tcp",
+			}},
+		}},
+	}
+
+	kubeClient := kubefake.NewSimpleClientset(emptyEndpoint, filledEndpoint)
+
+	discovery, _ := kubeClient.Discovery().(*fakediscovery.FakeDiscovery)
+	discovery.FakedServerVersion = &version.Info{
+		GitVersion: "v1.19",
+	}
+
+	client := newClientImpl(kubeClient)
+
+	stopCh := make(chan struct{})
+
+	eventCh, err := client.WatchAll(nil, stopCh)
+	require.NoError(t, err)
+
+	select {
+	case event := <-eventCh:
+		ep, ok := event.(*corev1.Endpoints)
+		require.True(t, ok)
+
+		assert.True(t, ep.Name == "empty-endpoint" || ep.Name == "filled-endpoint")
+	case <-time.After(50 * time.Millisecond):
+		assert.Fail(t, "expected to receive event for endpoints")
+	}
+
+	emptyEndpoint, err = kubeClient.CoreV1().Endpoints("test").Get(context.TODO(), "empty-endpoint", metav1.GetOptions{})
+	assert.NoError(t, err)
+
+	emptyEndpoint.Annotations["test-annotation"] = "___"
```

Comment:
Why do you change the annotation here? Why is this important?

Reply:
I wanted to simulate the changes to endpoints in my cluster that would previously trigger config reloads.
Those endpoints were used for leader election by updating the `control-plane.alpha.kubernetes.io/leader` annotation.
So this test will trigger an endpoint update without relevant changes. I want to make sure that these changes won't result in a new event in the event channel.

Reply:
Could we add a comment here like:

```
// Update endpoint annotation, but keep the resource version the same to show an update that should not trigger an update event on our eventCh.
```

Reply:
I added a slightly different comment, as the Resource Version is in fact updated (incremented from `1244` to `1245`).
I had to do this manually as the fake client doesn't seem to do it and the change detection logic i added will exit early if the resource version stays the same between updates.